src/hotspot/share/gc/g1/g1ParScanThreadState.cpp

```diff
@@ -60,6 +64,8 @@ G1ParScanThreadState::G1ParScanThreadState(G1CollectedHeap* g1h,
     _surviving_young_words(NULL),
     _surviving_words_length(young_cset_length + 1),
     _old_gen_is_full(false),
+    _partial_objarray_chunk_size(ParGCArrayScanChunk),
+    _partial_array_stepper(n_workers),
```

Comment:
See above discussion about naming and factoring.  The same stepper can support multiple array types if the chunk size is external to the stepper.

Reply:
It can, but I think I would prefer multiple "steppers" for that case. This is of course just a matter of taste and I'm fine with leaving the chunk size external.